runv.go

```diff
@@ -82,6 +86,73 @@ func getTtySize(outFd uintptr, isTerminalOut bool) (int, int) {
 	return int(ws.Height), int(ws.Width)
 }
 
+func getDefaultID() string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	return filepath.Base(cwd)
+}
+
+func removeState(podId, root string, sock net.Listener) {
+	os.RemoveAll(path.Join(root, podId))
+	sock.Close()
+}
+
+func saveState(vmId, podId, root string) (net.Listener, error) {
+	podPath := path.Join(root, podId)
+
+	_, err := os.Stat(podPath)
+	if err == nil {
+		return nil, fmt.Errorf("Container %s exists\n", podId)
+	}
+
+	err = os.MkdirAll(podPath, 0644)
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	pwd, err := filepath.Abs(".")
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	state := specs.State{
+		Version: specs.Version,
+		ID:      vmId,
+		Pid:     -1,
+		Root:    pwd,
+	}
+
+	stateData, err := json.MarshalIndent(&state, "", "\t")
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	err = ioutil.WriteFile(path.Join(podPath, "state.json"), stateData, 0644)
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	sock, err := net.Listen("unix", path.Join(podPath, "runv.sock"))
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	defer func() {
+		if err != nil {
+			os.RemoveAll(podPath)
+		}
+	}()
```

Comment:
I whould like to store more information in state.json, the podid can be got by directory name. I also consider to store containerid not vmid in state.json. Let's decide it later. BTW, the state.json of runc is not compatible with ocf right now.